ibm/resource_ibm_network_gateway_test.go

```diff
@@ -0,0 +1,138 @@
+package ibm
+
+import (
+	"fmt"
+	//"log"
+	"strconv"
+	//"strings"
+	//"time"
+	"errors"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	//"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	//"github.com/softlayer/softlayer-go/filter"
+	//"github.com/softlayer/softlayer-go/helpers/location"
+	//"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	//"github.com/softlayer/softlayer-go/session"
+	//"github.com/softlayer/softlayer-go/sl"
+)
+
+func TestAccIBMNetworkGateway_Basic(t *testing.T) {
+	var networkGateway datatypes.Hardware
+	//could also be datatypes.hardware
+
+	hostname := acctest.RandString(16)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
```

Comment:
How about a checking for the appliance destroy? You can check `testAccCheckIBMComputeBareMetalDestroy` in bare metal test